File: from-0201-to-0300/0206-reverse-linked-list/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head or not head.next:
            return head
        
        new_head = head
        while head.next:
            neck = head.next
            head.next = neck.next
            neck.next = new_head
            new_head = neck
            
        return new_head

File: from-0201-to-0300/0206-reverse-linked-list/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func reverseList(head *ListNode) *ListNode {
    if head == nil || head.Next == nil {
        return head
    }
    newHead := head
    var neck *ListNode
    for head.Next != nil {
        neck = head.Next
        head.Next = neck.Next
        neck.Next = newHead
        newHead = neck
    }
    return newHead
}